controllers/storagecluster/reconcile.go

```diff
@@ -651,9 +653,41 @@ func (r *StorageClusterReconciler) reconcilePhases(
 
 	}
 
+	// create a `ocs-metrics-exporter-ceph-auth` secret for metrics exporter
+	// create a cephclient and it will create the secret
+	err := r.createMetricsExporterCephClient(instance)
+	if err != nil {
+		r.Log.Error(err, "Failed to create ceph client for metrics exporter.")
+		return reconcile.Result{}, err
+	}
```

Comment:
You shouldn't be calling this function here, in ocs operator a request to reconcile a resource gos into the list of functions to handle reconciles. This also means that you need to move this code to one of the existing `ensure` packages or create a new one just for it